app/src/main/java/com/celzero/bravedns/util/Constants.kt

```diff
@@ -15,158 +15,182 @@ limitations under the License.
-        // modified below link post v053c release
-        const val APP_DOWNLOAD_LINK = "https://rethinkdns.com/download"
+        // The version tag value(response) for the update check (both on-device and app update)
+        // TODO: have two different response versions for blocklist update and app update
+        const val UPDATE_CHECK_RESPONSE_VERSION = 1
 
+        // meta data for on-device blocklist
+        data class OnDeviceBlocklistsMetadata(val url: String, val filename: String)
+
+        val ONDEVICE_BLOCKLISTS = listOf(
+            OnDeviceBlocklistsMetadata("https://download.rethinkdns.com/blocklists",
+                                       ONDEVICE_BLOCKLIST_FILE_TAG_NAME),
+            OnDeviceBlocklistsMetadata("https://download.rethinkdns.com/basicconfig",
+                                       ONDEVICE_BLOCKLIST_FILE_BASIC_CONFIG),
+            OnDeviceBlocklistsMetadata("https://download.rethinkdns.com/rank",
+                                       ONDEVICE_BLOCKLIST_FILE_RD_FILE),
+            OnDeviceBlocklistsMetadata("https://download.rethinkdns.com/trie",
+                                       ONDEVICE_BLOCKLIST_FILE_TD_FILE))
+
+        // url to download the rethinkdns apk file
+        const val RETHINK_APP_DOWNLOAD_LINK = "https://rethinkdns.com/download"
+
+        // base-url for bravedns
         const val BRAVE_BASE_STAMP = "https://basic.bravedns.com/"
-        const val BRAVE_CONFIGURE_BASE_STAMP = "rethinkdns.com/configure"
 
-        const val DOWNLOAD_STATUS_SUCCESSFUL = "STATUS_SUCCESSFUL"
+        // base-url stamp for configure blocklist
+        const val RETHINK_BLOCKLIST_CONFIGURE_BASE_STAMP = "rethinkdns.com/configure"
+
+        // base-url for bravedns
+        const val BRAVE_BASE_URL = "bravedns.com"
 
-        const val BRAVE_BASIC_URL = "bravedns.com"
-        const val RETHINK_BASIC_URL = "rethinkdns.com"
-        const val APPEND_VCODE = "vcode="
+        // base-url for rethinkdns
+        const val RETHINK_BASE_URL = "rethinkdns.com"
 
-        //constants for the server response json
+        // json object constants received as part of update check
+        // FIXME: Avoid usage of these parameters, map to POJO instead
         const val JSON_VERSION = "version"
         const val JSON_UPDATE = "update"
         const val JSON_LATEST = "latest"
 
-        //Firewall system components
+        // Firewall app category constants
+        // category: system components
         const val APP_CAT_SYSTEM_COMPONENTS = "System Components"
+
+        // category: system apps
         const val APP_CAT_SYSTEM_APPS = "System Apps"
+
+        // category: others
         const val APP_CAT_OTHER = "Other"
+
+        // category: non-app system
         const val APP_NON_APP = "Non-App System"
+
+        // category: installed apps
         const val INSTALLED_CAT_APPS = "Installed Apps"
+
+        // app type unknown
         const val UNKNOWN_APP = "Unknown"
```

Comment:
move these to `strings.xml`

and if used for db then move to categories-repository as enum?

Reply:
Moved to `CategoryInfoRepository` as enum class